### PR TITLE
Add Change Character App

### PR DESCRIPTION
**Pull request by daredoes**
Adds an app that changes the character to the hero in either Slot 1, 2, or 3.

### CODE REVIEW

File: src/main/java/bh/bot/app/ChangeCharacterApp.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                        sleep(mainLoopInterval);
+                        if (clickImage(BwMatrixMeta.Metas.Character.Dialogs.loading)) {
+                            debug("Character Loading!");
+                            characterLoaded = true;
+                            break;
+                        }
+                    } else {
+                        debug("Selecting Character in Slot #" + characterSlot);
+                        if (characterSlot == 1) {
+                            InteractionUtil.Mouse.mouseMoveAndClickAndHide(new Offset(250, 200).toScreenCoordinate());
```
Comment (9-9-9-9) [inline src/main/java/bh/bot/app/ChangeCharacterApp.java:93]: hardcoded offset

File: src/main/java/bh/bot/app/ChangeCharacterApp.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                        } else if (characterSlot == 2) {
+                            InteractionUtil.Mouse.mouseMoveAndClickAndHide(new Offset(375, 200).toScreenCoordinate());
```
Comment (9-9-9-9) [inline src/main/java/bh/bot/app/ChangeCharacterApp.java:95]: hardcoded offset

File: src/main/java/bh/bot/app/ChangeCharacterApp.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                        } else if (characterSlot == 3) {
+                            InteractionUtil.Mouse.mouseMoveAndClickAndHide(new Offset(500, 200).toScreenCoordinate());
```
Comment (9-9-9-9) [inline src/main/java/bh/bot/app/ChangeCharacterApp.java:97]: hardcoded offset

File: src/main/java/bh/bot/app/ChangeCharacterApp.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        final int characterSlot = arg;
+        info("Character in slot %3d ", characterSlot);
+        AtomicBoolean masterSwitch = new AtomicBoolean(false);
+        ThreadUtil.waitDone(
+                () -> doLoopClickImage(characterSlot, masterSwitch),
+                () -> internalDoSmallTasks( //
+                        masterSwitch, //
+                        SmallTasks //
+                                .builder() //
+                                .clickTalk() //
```
Comment (9-9-9-9) [inline src/main/java/bh/bot/app/ChangeCharacterApp.java:46]: are those options needed?

**Review by 9-9-9-9 — APPROVED** [review]

**Comment by 9-9-9-9** [comment]
Great job, tysm @daredoes 

Btw please pump version in `pom.xml` next time

**Comment by 9-9-9-9** [comment]
Btw @daredoes please write some text that I can paste into Wiki to introduce about your function

**Comment by daredoes** [comment]
@9-9-9-9 Will do! I'm also almost finished with an update for the AFK app that adds an infinite quest runner. My implementation works, but it could definitely use some love so users can customize their questing. It does support Boss Levels, Heroic Dungeons, and the next available level. It tries to search for a level in that order.

Combined with this PR, I have a script that goes
- AFK Run Character 1
- Change to Character 2
- AFK Run Character 2
- Change to Character 3
- AFK Run Character 3
- Change to Character 1
- Restart Actions from step 1 for 24/7 loop

File: src/main/java/bh/bot/app/ChangeCharacterApp.java (hunk 1 above)
Comment (daredoes) [reply src/main/java/bh/bot/app/ChangeCharacterApp.java:93]: These are hardcoded as there is no unique image in that location that can be recognized.

File: src/main/java/bh/bot/app/ChangeCharacterApp.java (hunk 4 above)
Comment (daredoes) [reply src/main/java/bh/bot/app/ChangeCharacterApp.java:46]: No clue! Didn't want to break anything :P I just started using your bot on Friday

**Comment by daredoes** [comment]
Wiki text:
function `character`

_This function only supports clicking the characters in the first, second, and third slot._

Supported flags:

 - `--exit=X` means bot will exit after X secs if not completed, no matter how many loop remaining
 - `--shutdown` shutdown your computer after bot exit
[Global flags](https://github.com/9-9-9-9/Bit-Heroes-bot/wiki/Application-flags)

Notes:

 - Automatically exit if used to select character that is already selected after a period of time

File: src/main/java/bh/bot/app/ChangeCharacterApp.java (hunk 1 above)
Comment (9-9-9-9) [reply src/main/java/bh/bot/app/ChangeCharacterApp.java:93]: The main purpose of Profile design is too support multiple screen resolution

File: src/main/java/bh/bot/app/ChangeCharacterApp.java (hunk 4 above)
Comment (9-9-9-9) [reply src/main/java/bh/bot/app/ChangeCharacterApp.java:46]: But you adopt the code base so fast, really impressive, ty

**Comment by 9-9-9-9** [comment]
> @9-9-9-9 Will do! I'm also almost finished with an update for the AFK app that adds an infinite quest runner. My implementation works, but it could definitely use some love so users can customize their questing. It does support Boss Levels, Heroic Dungeons, and the next available level. It tries to search for a level in that order.
> 
> Combined with this PR, I have a script that goes
> 
> * AFK Run Character 1
> * Change to Character 2
> * AFK Run Character 2
> * Change to Character 3
> * AFK Run Character 3
> * Change to Character 1
> * Restart Actions from step 1 for 24/7 loop

Great, please submit the script as well

**Comment by 9-9-9-9** [comment]
Hi @daredoes,

Look like due to lacking of dev guide in wiki, leads to some thing that does not perfect in the implementation, that's my fault. Hope that following comments is not too late :D

How to create TP image correctly
- When you are going to capture to crop a button, please take a picture of game screen via `sc` app.
- Go to `out/images/sc` folder and open that image using Paint
- You will see the image is **without any border**, a completely 800x520 screenshot
- Crop the button into another bmp image
- Import using `tp` app
- Now take a look at the newly imported `tp`, check the top left corner of that image, refer to the original image taken by `sc` app, take the coordinate into profile.
- Let say the Confirm button you made in this PR, as I checked, the offset should be 340,444 (my account does not have any NFT character)
- You can refer to my previously imported images, you will see the top left corner of the imported images has exact position as described in profile class, on the original image captured using `sc` app

**Pull request merged**